Add some ansible-galaxy documentation and fix subdir searching (#73406)
ansible/ansible: docs/docsite/rst/shared_snippets/installing_collections.txt

   ansible-galaxy collection install my_namespace-my_collection-1.0.0.tar.gz -p ./collections

> You can install a collection from a directory, much like ``pip``. The collection will be built with the ``MANIFEST.json`` or ``galaxy.yml``
> metadata in the directory.

Reviewer: ```suggestion
You can build and install a collection from a local source directory. The ``ansible-galaxy`` utility builds the collection using the ``MANIFEST.json`` or ``galaxy.yml``
metadata in the directory.
```

.. code-block:: bash

   ansible-galaxy collection install /path/to/collection -p ./collections

You can also install multiple collections in a namespace directory.

.. code-block:: text

>    ns
>    ├── collection1

Reviewer 2: I think it's better to use the `ls -Fl` format and add trailing slashes to dirs
```suggestion
   ns/
   ├── collection1/
```

Author: thanks!

   │   ├── MANIFEST.json

>    │   └── plugins
>    └── collection2

Reviewer 2: ```suggestion
   │   └── plugins/
   └── collection2/
```

       ├── galaxy.yml

>        └── plugins

Reviewer 2: ```suggestion
       └── plugins/
```
